### DISCUSSION TITLE
[Run issue]: Commissioning TOPAS to a certain proton machine using the Schneider convertor.

### DISCUSSION THREAD

**Discussion by tracyfhy** [OpenTOPAS/OpenTOPAS#82] [Run-Problem]
### Contact Details

tracyfhy2015@outlook.com

### What was the issue?

I am currently trying to commission TOPAS according to our proton machine, which requires the CT to be calibrated.
After surfing the google forum (more related topics and answers can be found there since OpenTOPAS is rather new), I understand that TOPAS is using some Schneider method with a density correction factor, which is used to make TOPAS consistent with a specific TPS and can be calculated by the ratio of RSP of the TPS and TOPAS ( I used the TOPAS extensions shared by Jan several years ago to calculate the TOPAS RSP, https://github.com/topasmc/extensions/tree/master/HU).

At our institute, there is another Opensource Monte Carlo dose engine already commissioned to our proton machine, which is MCsquare which also used the Schneider method for HU to density and material conversion.
The HU to density and HU to material are a little bit different though since we have made it machine specific.
Right now, I am frustrating with how to translate the commissioned HU to density and material conversion to TOPAS by using the schneider converter.

What I have tried is as follows:
1, get the RSP of TOPAS using your extension (I have tried both the default material and the one modified according to the material used by MCsquare which is attached below).
2, get the RSP of MCsquare (literally our TPS since it has been well commissioned) by:
  a, scaling the HU from -1000 to 2995;
     b, keeping the proton energy to be 100 MeV;
     c, getting the stopping power of the specific material which is scaled by the ratio between the density calculated by HU-to-density converter and the density of the material for every HU;
     d, getting the stopping power of water;
     e, getting the RSP.
3, get the density correction factor for each HU.

However, I still can't get the correct range using TOPAS to simulate a proton beam in water.

Does anyone have any comments/suggestions to what I have done?

The first attached file is the one I used when simulating proton beams, the second one is the one I generated based on the MCsquare HU to density and material converter with out density correction which I used to generate RSP for TOPAS.
[HUtoMaterialSchneider.txt](https://github.com/user-attachments/files/17179383/HUtoMaterialSchneider.txt)
[HUtoMaterialSchneiderNoCorrection.txt](https://github.com/user-attachments/files/17179384/HUtoMaterialSchneiderNoCorrection.txt)

Thanks a lot in advance!!

### TOPAS category

Physics

### Version

3.9

### What Operative System (OS) are you running on?

Linux

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### Attestation

- [X] I testify that I have fully read the bug reporting guideline beforehand.

**Comment by Anzalone394** [comment]
hello

I reached out to you in order to have a better chat after seeing your article.
Are you still in need of assistance? If so, please let me know so that we may have a more in-depth conversation.
message me on discord: bushraa03

**Reply by tracyfhy** [reply]
Sorry for the late replay.
I am still in need of help.

**Reply by Anzalone394** [reply]
ok cool sure i will happy to help you with it, did you have discord for better communication on it
send me your discord name
or add me on discord....   bushraa03

**Comment by jschuemann** [comment]
First off, I believe the reply by Anzalone394 looks like spam, so be careful if you discuss separately there, and sorry if that is not spam and a sincere attempt to help.

A lot of your steps seem to be correct, but I am a little confused by what you exactly do because the two HU files you share seem to be the same, except for the DensityCorrection weights. To commission the HU for your machine, you should:
1. Get the HU RSP curve from your TPS
2. Use the HU file with all DensityCorrection weights set to 1  and write out the stopping power for each HU using the extension
3. Divide the RSP from the TPS by the output RSPs, that should give you 3996 weights (close to 1). 
4. Replace the weights (DensityCorrection) in the HU file with the weights obtained
5. use the new HU file.

I hope this helps and gets the correct depths.
